src/Operators/finitedifference.jl

```diff
@@ -1760,6 +1759,89 @@ function stencil_right_boundary(
     bc.val
 end
 
+
+
+abstract type CurlFiniteDifferenceOperator <: FiniteDifferenceOperator end
+return_eltype(::CurlFiniteDifferenceOperator, arg) =
+    Geometry.curl_result_type(Val((3,)), eltype(arg))
+
+"""
+    C = CurlC2F(;boundaryname=boundarycondition...)
+    C.(v)
+
+Compute the vertical-derivative contribution to the curl of a center-valued
```

Comment:
```suggestion
Compute the vertical-derivative contribution to the horizontal curl of a center-valued
```

Reply:
I think adding "horizontal" to describing the curl is minimally needed to make this clear, we could also specify that the input vector and output vectors are also horizontal, but I think that is inferable.

Reply:
Is it really clearer? I'm actually a bit more confused now that I read it this way. Is the curl horizontal or the vectors it acts on?

Reply:
I would cal it the "vertical curl", in that it represents the vertical derivative contribution to the curl (in the same way that we define "vertical gradient" or "vertical divergence"). However it only acts on the horizontal covariant component (the vertical covariant component is in the null space of the operator), and it returns a horizontal contravariant  component.

Reply:
I think that the description will be clear so long we document the direction of the input and output vectors. This is the way I typically view these operators.

 - Gradient: scalar -> vector, scalar to one direction
 - Divergence: vector -> scalar, all directions to scalar
 - Curl: vector -> vector, output direction is normal to two input directions

Also, it might be nice to also write how this operator would be written in a more general context with projections. For example, we could include vector identities that we plan to enforce by composing this operator with other complementary differential operators.

I don't want to slow things down, so I'm fine to just merge and open an issue about improving the docs.

Reply:
I also think of these differential operators this way. Hence, since given two vectors, the curl is supposed to return a vector normal to the plane spanned by the two input vectors, and here we're only concerned with its vertical component derivatives, I'm a bit confused about why it returns a contravariant horizontal component. But yes, we should improve documentation all of our vector representations and operators

Reply:
Another way of thinking about it: if you think of the curl as the cross product of a vector with a differential operator, then you are just taking the cross product with the differential operator in the vertical direction. Since this acts in the vertical covariant direction, the cross product will only depend on the horizontal covariant components of the vector, and the result will have vertical contravariant component 0 (i.e. it will be a horizontal contravariant vector)

Reply:
I see, thanks for explaining it this way!